transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/core/MessagingClient.kt

```diff
@@ -25,13 +25,19 @@ interface MessagingClient {
          */
         object Connected : State()
 
+        /**
+         * Trying to reconnect after WebSocket failure,
+         */
+        object Reconnecting : State()
+
         /**
          * Session was successfully configured.
          *
          * @property connected true if session has been configured and connection is established.
          * @property newSession indicates if configured session is new. When configuring an existing session, [newSession] will be false.
+         * @property wasReconnecting indicates if connection is recovering.
          */
-        data class Configured(val connected: Boolean, val newSession: Boolean?) : State()
+        data class Configured(val connected: Boolean, val newSession: Boolean, val wasReconnecting: Boolean) : State()
```

Comment:
That might be a good idea.
I can think of 2 downsides to this approach.

1. In order to detect that we transition from Reconnecting state user will have to check for oldState == State.Reconnenting on **every** state change. While with current approach she/he only needs to make this check for State.Configured
2. That is going to be a breaking change. So we will need to deprecate current listener and add a new one. 
I was leaning towards something like that: 

`var onStateChanged: ((from: State, to: State) -> Unit)?`

Another option can be to add `data Reconnected(connected, newSession) : State` , but that feels a little bit weird as well, because of the argument names.

Reply:
Oh, I wasn't thinking they'd check for Reconnecting on every state change. I thought they would only be interested in checking if the old state was State.Reconnecting if the new state was State.Configured. If they'll have to check Reconnecting on every state change then I guess it's best to just stick with `wasReconnecting` in the Configured state.

For what it's worth, the reason I proposed containing the states in a class was to clearly differentiate the new and old states. Having the states as parameters like,

`var onStateChanged: ((from: State, to: State) -> Unit)?`

gets translated to the following on iOS:

`@property void (^ _Nullable stateListener)(GCMMessagingClientState *, GCMMessagingClientState *) __attribute__((swift_name("stateListener")));`

Which makes it impossible reading the code signature to know which of the new parameters is the new and which is the old state.

Reply:
I realized that there are no actual need to check for the Reconnecting state on every stateChange only after replying to your comment. :) It should be sufficient to check previous state only after receiving `Configure`.

That's a great catch. That you found that names are not translated. Going to check if there any way to enforce the names. But if not, data class is good solution.